client/src/components/Carousel.jsx

```diff
@@ -2,15 +2,16 @@ import React from 'react';
 import Listing from './Listing.jsx';
 import ListingsEnd from './ListingsEnd.jsx';
 
-const Carousel = ({ listings, like }) => {
+const Carousel = ({ listings, like, city }) => {
+
   return (
-  <div>
-    <ul>
-      {listings.map((listing, index) => <Listing key={index} like={like} listing={listing} />)}
-      <ListingsEnd />
-    </ul>
-  </div>
-)
+    <div>
+      <ul>
+        {listings.map((listing, index) => <Listing key={index} like={like} listing={listing} />)}
```

Comment:
Using index as the key works here, but I might use something that more specifically identifies the listing itself. I'm using it's id in the database. Probably just a preference thing though, depends on how you are utilizing the key.